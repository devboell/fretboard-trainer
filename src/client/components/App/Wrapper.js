import styled from 'styled-components'

const Wrapper = styled.div`
  width: 80%;
  margin: 10px 10%;
`

export default Wrapper
