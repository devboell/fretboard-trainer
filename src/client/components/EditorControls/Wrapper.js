import styled from 'styled-components'

const Wrapper = styled.div`
  height: 10%;
`

export default Wrapper
