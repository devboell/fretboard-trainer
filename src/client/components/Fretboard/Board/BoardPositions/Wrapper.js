import styled from 'styled-components'

const PositionWrapper = styled.div`
  display: flex;
  width: 100%;
  height: 100%;
`

export default PositionWrapper
