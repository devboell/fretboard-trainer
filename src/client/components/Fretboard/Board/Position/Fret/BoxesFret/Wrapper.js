import styled from 'styled-components'

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 80%;
  width: 90%;
  margin: 10% 5%;
  border: 1px solid darkgray;
  border-radius: 3px;
`

export default Wrapper
