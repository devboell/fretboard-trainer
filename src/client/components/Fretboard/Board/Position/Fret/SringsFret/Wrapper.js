import styled from 'styled-components'

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 5%;
  background-color: ${props => props.theme.background};
`

export default Wrapper
