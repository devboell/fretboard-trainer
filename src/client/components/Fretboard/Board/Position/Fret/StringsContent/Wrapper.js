import styled from 'styled-components'

const background = ({ theme, isHighlighted }) => (
  isHighlighted
    ? theme.highlight
    : theme.background
)

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 5%;
  background-color: ${props => background(props)};
`

export default Wrapper
