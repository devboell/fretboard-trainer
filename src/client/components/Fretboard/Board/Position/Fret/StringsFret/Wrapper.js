import styled, { css } from 'styled-components'

const highlightCSS = ({ theme }) => {
  const radiusPerc = 80
  const radius = Math.floor((theme.stringHeight / 100) * radiusPerc)

  return css`
    width: ${radius}px;
    height: ${radius}px;
    border-radius: 50%;
    background-color: ${theme.highlight};
  `
}


const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;

  ${props => (props.isHighlighted
    ? highlightCSS(props)
    : css`width: 100%;`)
} 
`

export default Wrapper
