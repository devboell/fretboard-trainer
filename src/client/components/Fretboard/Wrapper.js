import styled from 'styled-components'

const Wrapper = styled.div`
  width: 100%;
  height: ${props => props.height}px;
  background-color: ${props => props.theme.background};
`

/** @component */
export default Wrapper
