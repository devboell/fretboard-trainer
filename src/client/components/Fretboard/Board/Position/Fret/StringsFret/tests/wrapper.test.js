import React from 'react'
import theme from 'themes/fretboard-theme'
import 'jest-styled-components'

import Wrapper from '../Wrapper'

const defaultProps = {
  theme,
  isHighlighted: true,
}

describe('StringsFret Wrapper', () => {
  it('should return width 100%', () => {
    const wrapper = shallow(<Wrapper {...defaultProps} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should return highlighted circle', () => {
    const props = { ...defaultProps, isHighlighted: false }
    const wrapper = shallow(<Wrapper {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})

