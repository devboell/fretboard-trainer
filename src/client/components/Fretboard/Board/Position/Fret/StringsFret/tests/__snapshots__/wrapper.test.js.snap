// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StringsFret Wrapper should return highlighted circle 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

<div
  className="c0"
/>
`;

exports[`StringsFret Wrapper should return width 100% 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 32px;
  height: 32px;
  border-radius: 50%;
  background-color: orange;
}

<div
  className="c0"
/>
`;
